enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/AccessLoggingService.java

```diff
@@ -63,7 +66,11 @@ public void init() throws IOException {
         return new StreamObserver<>() {
             @Override
             public void onNext(StreamAccessLogsMessage message) {
-                AnalyticsFilter.getInstance().handleGRPCLogMsg(message);
+                if (ConfigHolder.getInstance().getConfig().getAnalyticsConfig().isEnabled()) {
```

Comment:
On next will not be called if the service is not enabled in envoy side, so we don't need the if condition in the on next method AFAIU

Reply:
Still there are two paths it can take right? depending on the configuration type it would decide whether analytics, metrics or both publishing needs to be done.

Reply:
Yes, we need to have the if condition here. We handle publishing analytics/metrics based on whether configuration is set to enable analytics, metrics or both.

Reply:
@Vathsan should not we have if condition rather than else if?

Reply:
Fixed in 45c855220c12a3238c90c7dad1f9a96afc80e427